### PR TITLE
catch a rejected promise

### PR DESCRIPTION
**Pull request by zpmorgan**
Hi!

The following code hangs due to an unresolved promise. I intentionally supplied a bad url.

```

const hostImageResolver = require('host-image-resolver');
   let da = 'https://raedrob.deviantart.com/art/Desert-Guardian-704243861';
   hostImageResolver(da + '5425423453254').then(urls => {
   console.log(urls);
 });

```

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -31,7 +31,7 @@ module.exports = function(url) {
         service(urlPieces).then((urls) => {
           // Covnvert to an array if we weren't passed one
           resolve(Array.isArray(urls) ? urls : [ urls ]);
-        });
+        }).catch( err => reject( 'service error: '+err))
```
Comment (dxprog) [inline index.js]: I think it's safe to just pass `reject` as the handler to `catch` by itself. Consumers can do with the error whatever they need. Also, semi-colon at the end please :)

**Review by dxprog — CHANGES_REQUESTED** [review]

File: index.js (hunk 1 above)
Comment (zpmorgan) [reply index.js]: hi dxprog. I've added the semicolon.

Dunno about the other bit. You mean change each service handler to use the initial `reject`? I'm not smart or motivated enough to do that :)

Comment (dxprog) [reply index.js]: Ah, the line would just become `.catch(reject);`. The only real difference is you'd get `err` without the `service error` part concatenated, which will be good if `err` is something other than a string. It doesn't really change the way you interact with it, but gives more flexibility.

Comment (zpmorgan) [reply index.js]: Okay, updated. Thanks for explaining. It makes sense now that I've thought about it.

**Review by dxprog — APPROVED** [review]

**Comment by dxprog** [comment]
Sweet. Thanks for fixing this!

**Pull request merged**